### PR TITLE
feat: allow team default for filtering test and internal data

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

In [this private slack thread](https://posthog.slack.com/archives/C03JJKLEC8L/p1661950826746349) a customer asked if you could set test and internal account filters on all new insights

This change allows that

## Changes

- [x] adds a new team setting to set the default
- [x] respects team setting when true over local default when false

## How did you test this code?

adding developer tests and running locally

### CODE REVIEW

**Review by Twixes — COMMENTED** [review]
This totally makes sense, though I'm a bit worried about project settings getting cluttered up 😅 At some point we'll need to organize this more

File: posthog/models/team/team.py [hunk 1]
```diff
@@ -109,7 +109,12 @@ class Team(UUIDClassicModel):
     signup_token: models.CharField = models.CharField(max_length=200, null=True, blank=True)
     is_demo: models.BooleanField = models.BooleanField(default=False)
     access_control: models.BooleanField = models.BooleanField(default=False)
+
     test_account_filters: models.JSONField = models.JSONField(default=list)
+    test_account_filters_default_checked: models.BooleanField = models.BooleanField(
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/models/team/team.py:114]: I believed we had a check that all added fields are `null=True`

(and was then relying on absence being falsey once we hit TS land)

[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/models/team/team.py:114]: (it's quick to check though!)

[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/models/team/team.py]: yep https://github.com/PostHog/posthog/runs/8131900643?check_suite_focus=true

File: frontend/src/scenes/insights/insightLogic.ts [hunk 2]
```diff
@@ -44,8 +44,9 @@ import { parseProperties } from 'lib/components/PropertyFilters/utils'
 const IS_TEST_MODE = process.env.NODE_ENV === 'test'
 const SHOW_TIMEOUT_MESSAGE_AFTER = 15000
 
-export const defaultFilterTestAccounts = (): boolean => {
-    return localStorage.getItem('default_filter_test_accounts') === 'true' || false
+export const defaultFilterTestAccounts = (current_filter_test_accounts: boolean): boolean => {
+    // if the current _gloabal_ value is true respect that over any local preference
```
Comment (Twixes) [inline frontend/src/scenes/insights/insightLogic.ts:48]: *global

File: frontend/src/scenes/project/Settings/TestAccountFiltersConfig.tsx [hunk 3]
```diff
@@ -37,6 +38,16 @@ export function TestAccountFiltersConfig(): JSX.Element {
...
                 )}
             </div>
+            <LemonSwitch
+                onChange={(checked) => {
+                    localStorage.setItem('default_filter_test_accounts', checked.toString())
+                    updateCurrentTeam({ test_account_filters_default_checked: checked })
+                }}
+                checked={!!currentTeam?.test_account_filters_default_checked}
+                disabled={currentTeamLoading}
+                label="Set this filter on all new insights"
```
Comment (Twixes) [inline frontend/src/scenes/project/Settings/TestAccountFiltersConfig.tsx:48]:
```suggestion
                label="Enable this filter on all new insights"
```

**Review by Twixes — APPROVED** [review]
Wording suggestions but overall looks good

**Pull request merged**